core/io/image.cpp

```diff
@@ -2567,15 +2567,14 @@ void Image::blit_rect_mask(const Ref<Image> &p_src, const Ref<Image> &p_mask, co
 
 	for (int i = 0; i < dest_rect.size.y; i++) {
 		for (int j = 0; j < dest_rect.size.x; j++) {
-			int src_x = clipped_src_rect.position.x + j;
-			int src_y = clipped_src_rect.position.y + i;
+			Point2i point = Point2i(j, i);
+			Point2i source = clipped_src_rect.position + point;
```

Comment:
I wouldn't play with the devil here, just like with other changes in `Image`. This is ok for the rest of the engine where image is used if proved to be easier to read and whatnot, but again `int` would likely perform faster.